src/ralph/admin/tests/tests_functional.py

```diff
@@ -200,6 +203,100 @@ class FooAdmin(RalphAdmin):
         tabs = response.context['change_views']
         for tab in tabs:
             resp = self.client.get(
-                reverse(tab.url_with_namespace, args=[obj.pk,])
+                reverse(tab.url_with_namespace, args=[obj.pk, ])
             )
             self.assertEqual(resp.status_code, 200)
+
+
+class ChangeListTest(TestCase):
+    def _change_list_factory(
+        self, model, model_admin, request, list_display=None
+    ):
+        return RalphChangeList(
+            date_hierarchy=model_admin.date_hierarchy,
+            list_display=list_display or model_admin.list_display,
+            list_display_links=model_admin.list_display_links,
+            list_editable=model_admin.list_editable,
+            list_filter=model_admin.list_filter,
+            list_max_show_all=model_admin.list_max_show_all,
+            list_per_page=model_admin.list_per_page,
+            list_select_related=model_admin.list_select_related,
+            search_fields=model_admin.search_fields,
+            model=model,
+            model_admin=model_admin(Car, ralph_site),
+            request=request,
+        )
+
+    def _get_ordering_list(self, model, model_admin, params, list_display):
+        from django.contrib.admin.views.main import ORDER_VAR
+        get_params = '{}={}'.format(ORDER_VAR, '.'.join(map(str, params)))
+        request = RequestFactory().get('/?' + get_params)
+        cl = self._change_list_factory(
+            model=Car,
+            model_admin=CarAdmin,
+            request=request,
+            list_display=list_display
+        )
+        return cl.get_ordering(request, model.objects.all())
+
+    def test_sort_related(self):
+        ordering = self._get_ordering_list(
+            Car, CarAdmin, [1, 2], ['pk', 'name', 'manufacturer']
+        )
+        test_data = [
```

Comment:
consider using [ddt](http://ddt.readthedocs.org/en/latest/index.html) for such a test